### ISSUE TITLE
New node added shift out of place

### ISSUE THREAD

**Issue by VincentK94** [dabeng/OrgChart#401]
Hi everyone! I have an issue when trying to add a new node. At first I suspect that it is due to other javascript / css files, but when I do a clean copy of edit-chart.html and its related javascript and css files. It still didn't work. 

I reclone and tried running edit-chart.html. It work, but after copying to my project, similar issue appear.

**After page loaded**
![screen shot 2018-07-05 at 9 14 23 pm](https://user-images.githubusercontent.com/34161211/42325710-9ddbcd1e-8099-11e8-8a44-f189cedfed4e.png)

**After adding a child node**
![screen shot 2018-07-05 at 9 14 36 pm](https://user-images.githubusercontent.com/34161211/42325716-a1499b70-8099-11e8-9694-15bcd00e3ca0.png)

**After adding a sibling node**
![screen shot 2018-07-05 at 9 14 51 pm](https://user-images.githubusercontent.com/34161211/42325718-a36c0b2c-8099-11e8-83bd-d543c695d7d6.png)

**Comment by mp93** [comment]
I have the same issue. If a node has a children on adding a new child node the parent node is shifting. Can someone help with this issue?

**Comment by babatopeoni** [comment]
Please is the any fix for this?

**Comment by sankhaKarunasekara** [comment]
does anyone have a solutions to this yet?

**Comment by CaptainCannabis** [comment]
Can anyone set up a fiddle? Can't imagine what should be the problem.

**Comment by sankhaKarunasekara** [comment]
@CaptainCannabis Hi, 

when I try to add new children to a node which already have a children or add sibling to a node the chart becomes what @VincentK94 described at the top. 

Thing is this works fine with no issues in an isolated environment. once I add the implantation into my project issue came up. so I don't thing fiddle will help.

I think issue is with `addSiblings: function($node, data) {` method. I narrow it down to 

 `this.complementLine($siblingTds.eq(insertPostion).after($nodeChart.closest('td').unwrap()), siblingCount, 1);`

line by comparing difference in local set up and my project. I suspect issue is with `$nodeChart.closest('td').unwrap()` this.

**Comment by CaptainCannabis** [comment]
The nodes are centered by using text-align. Maybe by editing your html sth regarding textflow is going wrong. You may try to change the css for the nodes and use margin for example:

```
div.node {
    width: 240px;
    display: block;
    padding: 0 3px;
    margin: 0 auto;
}
```

Like you said - something in youre enviroment is causing this. So a link to that would be helpfull :)

**Comment by sankhaKarunasekara** [comment]
I wish I could. sadly I am working on a production app. =D Thanks for the tip mate. I will try this.

**Comment by CaptainCannabis** [comment]
Maybe you could also try to check if the inserted elements are still expanding to the full width, so that text-align center would work. Most times there is a css conflict causing this behavieur.

"The text-align property specifies the horizontal alignment of text in an element."

So your element has to have a width greater than text length...

**Comment by sankhaKarunasekara** [comment]
Yep. It could be but what's bugging me is when I use drag and drop feature, It works perfectly. There is no miss alignments. If this is a css issue, that could also affect the drag and drop right?

**Comment by CaptainCannabis** [comment]
I dont know exactly if both methods are using the same way of inserting the modified html. The width of the columns depends on the table-layout which is used. Maybe after inserting youre table is not using the same css styles like before?

**Comment by sankhaKarunasekara** [comment]
Thanks for your help. @CaptainCannabis . We finally figured it out..! Instead of inserting <tr> tags directly to table as it suppose to some library in our app creating <tbody> inside <table> and then insert <tr> tags. so then the jQuery selector is not able identify the element that it wants. That was the issue. Thanks mate. you have been a big help. 👊👍

**Comment by CaptainCannabis** [comment]
> Thanks for your help. @CaptainCannabis . We finally figured it out..! Instead of inserting tags directly to table as it suppose to some library in our app creating inside
> 
> and then insert tags. so then the jQuery selector is not able identify the element that it wants. That was the issue. Thanks mate. you have been a big help. 👊👍

Yeah! Glad to hear that - most times people you just need a hint to find the right approach or mistake. So you're welcome! Greetings from germany and have a nice weekend! 👊
